addons/stock/models/stock_warehouse.py

```diff
@@ -29,7 +29,11 @@ class Warehouse(models.Model):
     # namedtuple used in helper methods generating values for routes
     Routing = namedtuple('Routing', ['from_loc', 'dest_loc', 'picking_type', 'action'])
 
-    name = fields.Char('Warehouse', index=True, required=True, default=lambda self: self.env.company.name)
+    def _default_name(self):
+        count = self.env['stock.warehouse'].search_count([('company_id', '=', self.env.company.id)])
```

Comment:
```suggestion
        count = self.env['stock.warehouse'].with_context(active_test=False).search_count([('company_id', '=', self.env.company.id)])
```

Reply:
makes sense with constraint, even if name is likely to be changed
thanks